arbitrum-docs/stylus/reference/rust-sdk-guide.md

```diff
@@ -408,29 +408,140 @@ interface Weth is Erc20 {
 
 ## Calls
 
-:::caution UNDER CONSTRUCTION
-
-This section is currently under construction, and will be updated soon.
-
-If you're waiting for this content to be completed, click the `Request an update` button at the top of this page to let us know!
+Just as with storage and methods, Stylus SDK calls are Solidity ABI equivalent. This means you never have to know the implementation details of other contracts to invoke them. You simply import the Solidity interface of the target contract, which can be auto-generated via the [`cargo stylus`] [CLI tool][abi_export].
 
+:::tip
+You can call contracts in any programming language with the Stylus SDK.
 :::
 
 ### [`sol_interface!`][sol_interface]
 
-_Coming soon!_
+This macro defines a `struct` for each of the Solidity interfaces provided.
+
+```rust
+sol_interface! {
+    interface IService {
+        function makePayment(address user) payable returns (string);
+        function getConstant() pure returns (bytes32)
+    }
+
+    interface ITree {
+        // other interface methods
+    }
+}
+```
+
+The above will define `IService` and `ITree` for calling the methods of the two contracts.
+
+For example, `IService` will have a `make_payment` method that accepts an [`Address`][Address] and returns a [`B256`][B256].
+
+```rust
+pub fn do_call(&mut self, account: IService, user: Address) -> Result<String, Error> {
+    account.make_payment(self, user)  // note the snake case
+}
+```
+
+Observe the casing change. [`sol_interface!`][sol_interface] computes the selector based on the exact name passed in, which should almost always be `CamelCase`. For aesthetics, the rust functions will instead use `snake_case`.
+
+### Configuring Gas and Value with [`Call`]
+
+[`Call`][Call] lets you configure a call via optional configuration methods. This is similar to how one would configure opening a [`File`][File] in Rust.
+
+```rust
+pub fn do_call(account: IService, user: Address) -> Result<String, Error> {
+    let config = Call::new()
+        .gas(evm::gas_left() / 2)       // limit to half the gas left
+        .value(msg::value());           // set the callvalue
+
+    account.make_payment(config, user)
+}
+```
+
+By default [`Call`] supplies all gas remaining and zero value, which often means `Call::new()` may be passed to the method directly. Additional configuration options are available in cases of reentrancy.
+
+### Reentrant Calls
```

Comment:
```suggestion
### Reentrant calls
```

Sentence-casing header text to maintain consistency with other docs ([style guide](https://www.notion.so/arbitrum/Pattern-guide-Docs-dafaf8d1ce34433d88c1ef3b4b99c19c?pvs=4#b420b2b1ff614c71b01743e1b140ffbe))